app/scripts/models/account.js

```diff
@@ -1281,6 +1281,34 @@ define(function (require, exports, module) {
       return this._fxaClient.checkTotpTokenExists(
         this.get('sessionToken')
       );
+    },
+
+    /**
+     * Consume a recovery code.
+     *
+     * @param {String} code
+
+     * @returns {Promise}
+     */
+    consumeRecoveryCode (code) {
+      return this._fxaClient.consumeRecoveryCode(
+        this.get('sessionToken'),
+        code,
+        {
+          metricsContext: this._metrics.getFlowEventMetadata(),
```

Comment:
@vbudhram I guess one question is, further to our conversation yesterday, do we actually need to send `metricsContext` to this endpoint or will it already be stored in memcached with the session token?

Obviously there's no real harm in sending it again but for simplicity reasons I quite like the idea of only the "entrypoint" routes being the ones that set it, with everything else reading it until the complete event is triggered. And sorry for not raising this question before, when you were adding `metricsContext` to the js-client and auth-server PRs. 😕

Reply:
> I wonder if it might be worthwhile to do an audit of existing routes to make sure `metricsContext` is set correctly when it needs to be.

It probably is, I realise I haven't done a good job of communicating what the right thing to do is with `metricsContext` generally. If we're going to do that anyway, you can leave it in here if you like and it will fall out of the wider sweep when we do that.